core/src/main/java/org/apache/accumulo/core/rpc/clients/TServerClient.java

```diff
@@ -155,18 +194,22 @@ default Pair<String,C> getThriftServerConnection(Logger LOG, ThriftClientTypes<C
       throw new UncheckedIOException("Error creating transport to debug host: " + debugHost
           + ". If this server is down, then you will need to remove or change the system property "
           + DEBUG_HOST + ".", new IOException(""));
+    } else if (debugRG != null && rgp.test(ResourceGroupId.of(debugRG))) {
```

Comment:
Not sure about this, but the code may be a bit more maintainable if a boolean is set when the debugRG was actually used.  Then this code does not need to duplicate the code that decides when to use and have to be kept in sync w/ the earlier code.  Also the code that decides to use it or not seems more complicated than this.  Could do the same pattern for the debug host.

```suggestion
    } else if (usedDebugRG) {
```